processing/src/main/java/io/druid/segment/StringDimensionMergerV9.java

```diff
@@ -294,6 +295,7 @@ public void writeIndexes(List<IntBuffer> segmentRowNumConversions, Closer closer
     try (
         Closeable toCloseEncodedValueWriter = encodedValueWriter;
         Closeable toCloseBitmapWriter = bitmapWriter;
+        Closeable toCloseDictionaryMergeIterator = dictionaryMergeIterator;
```

Comment:
Do we need to wait until here to close the iterator? Could it be closed in `writeMergedValueMetadata`?